### PR TITLE
Be able to have multiple incomes per contract #9402

### PR DESCRIPTION
**Pull request by phpgeekfr**
Be able to have multiple incomes per contract #9402

### CODE REVIEW

File: libs/model/src/lib/income.ts [hunk 1]
```diff
@@ -1,13 +1,41 @@
...
   termId: string;
-  /** The contract (sale in this case) that created this income */
+  /** The contract that created this income */
   contractId: string;
   price: number;
   currency: MovieCurrency;
   offerId?: string;
+  titleId?: string;
+  status: 'pending' | 'processed'
+  date: Date
```
Comment (phpgeekfr) [inline libs/model/src/lib/income.ts:16]:
```suggestion
  status: 'pending' | 'processed';
  date: Date;
```

File: libs/import/src/lib/view-extracted-elements/contract/view-extracted-contracts.component.ts [hunk 2]
```diff
@@ -3,7 +3,6 @@ import { Component, ChangeDetectionStrategy, OnInit, Input, Inject } from '@angu
 import { BehaviorSubject, firstValueFrom } from 'rxjs';
```
Comment (n-germain-work) [inline libs/import/src/lib/view-extracted-elements/contract/view-extracted-contracts.component.ts:3]: 1st line is empty

File: libs/import/src/lib/view-extracted-elements/organizations/utils.ts [hunk 3]
```diff
@@ -59,7 +59,7 @@ export async function formatOrg(sheetTab: SheetTab, organizationService: Organiz
   const fieldsConfig: FieldsConfigType = {
     /* a */ 'org.name': async (value: string) => {
       if (!value) throw mandatoryError(value, 'Organization Name');
-      const exist = await getOrgId(value, organizationService, orgNameCache);
+      const exist = await getOrgId(value, organizationService, orgNameCache, centralOrg);
       if (exist) throw alreadyExistError(value, 'Organization Name');
       return value
```
Comment (n-germain-work) [inline libs/import/src/lib/view-extracted-elements/organizations/utils.ts:64]: missing `;`

File: libs/import/src/lib/view-extracted-elements/organizations/view-extracted-organizations.component.ts [hunk 4]
```diff
@@ -10,6 +10,7 @@ import { OrganizationService } from '@blockframes/organization/service';
```
Comment (n-germain-work) [inline libs/import/src/lib/view-extracted-elements/organizations/view-extracted-organizations.component.ts:10]: 1st line at the top is empoty

File: libs/model/src/lib/avail.ts [hunk 5]
```diff
@@ -577,6 +571,45 @@ export function territoryAvailabilities({
...
+  existingSales: FullSale[],
+  existingMandates?: FullMandate[]
+}
+
+export function isTermOverlappingExistingContracts({ term, existingSales, existingMandates = [] }: OverlappingOptions): { licensed: boolean, sold: boolean } {
+
+  assertValidTitle(existingMandates, existingSales, []);
+
+  const mandates: Territory[] = [];
+  const sales: Territory[] = []
```
Comment (n-germain-work) [inline libs/model/src/lib/avail.ts:585]: missing `;`

File: libs/model/src/lib/income.ts [hunk 6]
```diff
@@ -1,13 +1,41 @@
...
+}
+
+export interface TotalIncome {
+  EUR: number;
+  USD: number;
+}
```
[the remark this answers is not in the document]
Comment (phpgeekfr) [reply libs/model/src/lib/income.ts:22]: created a dedicated issue #9422

File: libs/import/src/lib/utils.ts [hunk 7]
```diff
@@ -123,9 +123,13 @@ export async function getContract(
...
   if (cache[id]) return cache[id];
 
-  const contract = await contractService.getValue(id);
-  cache[id] = contract;
-  return contract;
+  try {
+    const contract = await contractService.getValue(id);
+    cache[id] = contract;
+    return contract;
+  } catch (err) {/**do nothing*/ }
```
[the remark this answers is not in the document]
Comment (phpgeekfr) [reply libs/import/src/lib/utils.ts:130]: Here, we want to check if `id` used in Excel file belongs to an existing contract or not. If id does not belongs to a contract in DB, we don't want to raise an error

**Pull request merged**